CaesarCipher/CaesarCipher/main.m

```diff
@@ -10,42 +10,287 @@
 
 @interface CaesarCipher : NSObject
 
-- (NSString *)decode:(NSString *)string offset:(int)offset;
-- (NSString *)encode:(NSString *)string offset:(int)offset;
+- (NSString *)decode:(NSString *)string offset:(NSInteger)offset;
+- (NSString *)encode:(NSString *)string offset:(NSInteger)offset;
+
+- (BOOL)easyIsSameMessage:(NSString *)msg1 asMessage:(NSString *)msg2;
+- (BOOL)hardIsSameMessage:(NSString *)msg1 asMessage:(NSString *)msg2;
 
 @end
 
-@implementation CaesarCipher
+@implementation CaesarCipher {
+    NSInteger _alphabetSize;
+}
+
+-(id)init {
+    if (self = [super init]) {
+        _alphabetSize = 26;
+    }
+    return self;
+}
 
-- (NSString *)encode:(NSString *)string offset:(int)offset {
-    if (offset > 25) {
-        NSAssert(offset < 26, @"offset is out of range. 1 - 25");
+- (NSString *)encode:(NSString *)string offset:(NSInteger)offset {
+    
+    // if offset is an illegal value, the assert statement will
+    // fail and the app will crash with the provided error message
+    if (offset > _alphabetSize - 1) {
+        NSAssert(offset < _alphabetSize, @"offset is out of range. 1 - 25");
```

Comment:
Suggestion: If you're going to store _alphabetSize in a variable I wouldn't do it as a property. It's not a defining characteristic of the Cipher itself, it's just a piece of information that the Cipher will use to execute correctly. My suggestion is to make a *constant* at the top of your file. You can do this with the `#define` keyword.

```objective-c
#define ALPHABET_SIZE 26
```

By convention, constance are all caps. They also go at the top of your file.

```objective-c
#import <Foundation/Foundation.h>

#define ALPHABET_SIZE 26

@interface CaesarCipher : NSObject
```